packages/components/toast/src/toast.tsx

```diff
@@ -87,7 +87,7 @@ export function createRenderToast(
   const { render, toastComponent: ToastComponent = Toast } = options
   const renderToast: React.FC<RenderProps> = (props) => {
     if (typeof render === "function") {
-      return render(props) as JSX.Element
+      return render({ ...props, options }) as JSX.Element
```

Comment:
What we want here is for the options to be inlined, not as a dedicated key


```suggestion
      return render({ ...props, ...options }) as JSX.Element
```